05_iigo/index.tex

```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
     \item This information is aggregated into a format accepted by the interface of presidential functions (usually in form $map[agent] \xrightarrow[]{} information$).
     \item Presidential function is called only on the currently residing President with aggregated information as inputs.
-    \item The outcome of the presidential function call is processed.
+    \item The outcome of the presidential function call is processed. This includes updating the game state, logging information for monitoring, subtracting action costs.
     \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
 \end{enumerate}
 
 \subsection{Legislative Branch}
+Similarly to the president the Speaker also has client-side functions which directly map to the Speaker powers described in \ref{sec:legislative}, as well as server-side functions which not only facilitate the implementation of these powers but also handle other functionality, such as the logging needed for monitoring and withdrawal of costs associated with the power.
+
+\subsubsection{Client-side}
+The legislative branch functions can be divided into two categories:
+\begin{enumerate}
+\item Speaker power functions:
+    \begin{itemize}
+        \item \texttt{DecideAgenda} does not directly map to a power of the speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
+        \item \texttt{DecideVote} enables the institutional power of calling a vote and deciding the participating islands. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and a list of all the alive island IDs. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker is the rule that is passed to agents returned by the Speaker if a vote is to be conducted as indicated by the boolean returned by the Speaker.
+        \item \texttt{DecideAnnouncement} enables the institutional power of declaring a result of a vote on rules. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and the result of the vote held. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker along with the result returned by the Speaker is the information that is processed by the server and broadcasted to all the agents if the agent has chosen to make a declaration as indicated by the boolean returned by the Speaker.
```

Comment:
result of a vote on rules/??
 can you revise this plz , not very understandable